SWIPs/swip-multiple_payment_processing_support.md

```diff
@@ -130,27 +249,107 @@ type Peer struct {
 }
 ```
 
-One option is to move the accounting responsibilities from ```Swap``` to a new component (from now on ```Accounting```), introduce it as a new member of the ```Peer``` struct and add a new collaborator on which we can delegate the payment processing. This collaborator (from now on ```SwarmPayments```) will provide access to the supported payment modules, being Swarm one of such modules. This design decouples the accounting from the payment processing. We refer to the concrete payment module implementations as a ```PaymentProcessor```s.
+One option is to move the accounting responsibilities from ```Swap``` to a new component (from now on ```Accounting```), introduce it as a new member of the ```Peer``` struct and add a new collaborator on which we can delegate the payment processing. This collaborator (from now on ```SwarmPayments```) will provide access to the supported payment modules, being ```Swap``` one of such modules. This design decouples the accounting from the payment processing. We refer to the concrete payment module implementations as a ```PaymentProcessor```s.
 
 ```golang
 // Peer is a devp2p peer for the Swap protocol
 type Peer struct {
 	*protocols.Peer
 	accounting         *Accounting
 	payments           *SwarmPayments
-	backend            contract.Backend
-	beneficiary        common.Address
-	contractAddress    common.Address
-	lastReceivedCheque *Cheque
 }
 ```
 
-The ```payments``` member of the ```Peer``` struct holds the ```SwarmPayments``` component described previously, which is responsible to hold the particular ```PaymentProcessor``` implementations supported by the node and a mapping of:
+This option will severely change the current architecture of Swap and affect how accounting works. A better approach to preserve the accounting structure as defined in ```p2p/protocols/accounting.go``` and its related modules is to keep the relationship between the Swap ```Peer``` and the ```Swap``` accounting object and introduce the ```SwarmPayments``` object as a collaborator of ```Swap```. This way we keep the accounting structure as it is today and delegate all payment related logic from ```Swap``` to the corresponding payment module via  ```SwarmPayments```. Following this idea, all the payment related members from the ```Swap``` object should be moved to the Swap payment module implementation.
```

Comment:
I think "severely" is not needed here.

Reply:
Well it actually is correct, the first option actually affects severely the current architecture, but you could also use some other word, like "importantly" or whatever